### ISSUE TITLE
Install/build pkg-config if it's not available on the system

### ISSUE THREAD

**Issue by sdroege** [rust-lang/pkg-config-rs#103]
Shouldn't be too hard via the `cc` crate and downloading it in `build.rs`.

**Comment by lu-zero** [comment]
Potentially [libpkgconf](https://github.com/pkgconf/pkgconf/tree/master/libpkgconf) could be wrapped in a -sys crate but even that is a bit large even if self-contained.

**Comment by Be-ing** [comment]
> Potentially [libpkgconf](https://github.com/pkgconf/pkgconf/tree/master/libpkgconf) could be wrapped in a -sys crate but even that is a bit large even if self-contained.

I had the same idea.

**Comment by nirbheek** [comment]
There's also a [meson port](https://gitlab.freedesktop.org/tpm/pkg-config) for the original pkg-config now, which is lighter than pkgconf.

**Comment by lu-zero** [comment]
the meson port removed glib from the codebase? IIRC that's one of the issues with the original pkg-config.

**Comment by nirbheek** [comment]
> the meson port removed glib from the codebase? IIRC that's one of the issues with the original pkg-config.

No, that was Jussi's meson port. This one by @tp-m uses glib as a subproject, with all unnecessary things patched out in the build system, see https://gitlab.freedesktop.org/tpm/pkg-config/-/tree/meson/subprojects/packagefiles/glib-2.74.4. That makes it equivalent to how the Autotools build works, except with newer glib.

**Comment by lu-zero** [comment]
Then there is lots of room for making it light :) I wish enough of us has time to come up with a pure rust impl.
